static/js/confirm_dialog.js

```diff
@@ -29,8 +30,22 @@ import * as overlays from "./overlays";
            only ever have one confirm dialog active at any
            time.
 
+        6) The loading spinner, if required, must be handled by the
+           caller.
```

Comment:
This is unclear.  I'd be more explicit about what the correct interface is, e.g.

"If a modal wants a loading spinner, it should pass `loading_spinner: true` and manage the
spinner via calling the functions `hide_confirm_dialog_spinner` and `show_confirm_dialog_spinner` 
via its callback hooks", or something like that.